src/NSwag.CodeGeneration.CSharp/Templates/ControllerTemplate.tt

```diff
@@ -15,47 +13,62 @@ public interface I<#=Model.Class#>Controller
 {
-    private I<#=Model.Class#>Controller _implementation; 
+	<# if (Model.GenerateControllerInterfaces){#>
+    private I<#=Model.Class #>Controller _implementation; 
 
-    public <#=Model.Class#>Controller(I<#=Model.Class#>Controller implementation)
+    public <#=Model.Class #>Controller(I<#=Model.Class #>Controller implementation)
     {
         _implementation = implementation; 
     }
-
-<#foreach(var operation in Model.Operations){#>
+	<#}#><#foreach(var operation in Model.Operations){#>
 <#  if(operation.HasSummary){#>    /// <summary><#=ConversionUtilities.ConvertCSharpDocBreaks(operation.Summary, 1)#></summary>
 <#  }foreach (var parameter in operation.Parameters){#>
 <#      if(parameter.HasDescription){#>    /// <param name="<#=parameter.VariableName#>"><#=ConversionUtilities.ConvertCSharpDocBreaks(parameter.Description, 1)#></param>
 <#  }}if(operation.HasResultDescription){#>    /// <returns><#=ConversionUtilities.ConvertCSharpDocBreaks(operation.ResultDescription, 1)#></returns>
-<#  }if(operation.IsDeprecated){#>    [System.Obsolete]
-<#  }#>    [System.Web.Http.Http<#=operation.HttpMethodUpper#>, System.Web.Http.Route("<#=operation.Path#>")]
-    public <#if(Model.WrapResponses){#>async System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage><#}else{#><#=operation.ResultType#><#}#> <#=operation.ActualOperationName#>(<#foreach(var parameter in operation.Parameters){#><#=parameter.Type#> <#=parameter.VariableName#><#if(Model.GenerateOptionalParameters && parameter.IsOptional){#> = null<#}#><#if(!parameter.IsLast){#>, <#}}#>)
-    {    
+<#  }if(operation.IsDeprecated){#>    [System.Obsolete]<#}#>
+	[System.Web.Http.Http<#=operation.HttpMethodUpper#>, System.Web.Http.Route("<#=operation.Path#>")]
+	public <# if(Model.GenerateAbstractControllers){#>abstract <#}#> <# if(Model.WrapResponses){#><# if(!Model.GenerateAbstractControllers){#>async <#}#>System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage><#}else{#><#=operation.ResultType#><#}#> <#=operation.ActualOperationName#>(<#foreach(var parameter in operation.Parameters){#><#=parameter.Type#> <#=parameter.VariableName#><# if(Model.GenerateOptionalParameters && parameter.IsOptional){#> = null<#}#><# if(!parameter.IsLast){#>, <#}}#>)<# if(Model.GenerateAbstractControllers){#>;
+	<#}else {#>
+    {  
+	<#}#>
+	<# if(Model.GenerateControllerInterfaces){#>
 <#          if (Model.WrapResponses){#>
-        var result = await _implementation.<#=operation.ActualOperationName#>Async(<#foreach(var parameter in operation.Parameters){#><#=parameter.VariableName#><#if(!parameter.IsLast){#>, <#}}#>);
+        var result = await _implementation.<#=operation.ActualOperationName#>Async(<#foreach(var parameter in operation.Parameters){#><#=parameter.VariableName#><# if(!parameter.IsLast){#>, <#}}#>);
 
```

Comment:
You can probably revert formatting changes on this line and on lines 48 and 55 as well